up-l1/README.adoc

```diff
@@ -311,6 +318,34 @@ end
 deactivate T
 ----
 
+== LocalUriProvider
+
+A uEntity can use the `LocalUriProvider` to create URIs representing the uEntity's local resources during runtime. This information can then be used in messages to be sent to other uEntities.
+
+A `UTransport` implementation can use the `LocalUriProvider` to determine the uEntity's authority during runtime. This information can might be useful for normalizing _local_ URIs passed into the Transport Layer API methods with authority information.
+
+=== GetAuthority
+
+A uEntity invokes this method to get its own authority.
+
+Implementations *MAY* use any appropriate mechanism to determine the local authority during runtime, e.g. by means of a configuration file, environment variables or a central registry.
+
+=== GetSource
+
+A uEntity invokes this method to get the address that it expects incoming Notification or RPC Response messages to be sent to.
+
+The address returned by an implementation *MUST* consist of the uEntity's (fixed) _authority_, _identifier_ and _major version_ and _resource ID_ `0x0000`.
+
+Implementations *MAY* use any appropriate mechanism to determine these values during runtime, e.g. by means of a configuration file, environment variables or a central registry.
+
+=== GetResource
```

Comment:
What code generators? ;-) Unless you pop them out of your hat soon, I guess we will need to be able to implement uServices etc. without them for a while, won't we?

Reply:
They are coming. We finished implementing them internally (for some languages) and started the process of open sourcing them (stripping out the stuff that doesn't belong in open source). I really was hoping to get help from the community but that didn't happen :-(. I'll dump them in up-tools so we can expand on them for the other programming languages (it might only support C++ and Java to start).

Reply:
Having said that, I'll remove my -1 and we can change it later as needed)